### PR TITLE
[CBRD-23746] Error occurs when joining using nvl() function based index.

### PR DESCRIPTION
**Pull request by shparkcubrid**
http://jira.cubrid.org/browse/CBRD-23746

Index key value is set incorrectly in XASL generation.
Function-based functions omitted from validate_regu_key_function_index() are added based on pt_expr_is_allowed_as_function_index().

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/parser/parse_tree_cl.c (1 incorrect indentation(s) found)
```
 18366       /* if add it here, add it to validate_regu_key_function_index () as well */
```
#### File: src/parser/xasl_generation.c (1 incorrect indentation(s) found)
```
 24188 	  /* if add it here, add it to pt_expr_is_allowed_as_function_index() as well */
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

File: src/parser/xasl_generation.c [hunk 1]
```diff
@@ -24181,6 +24181,7 @@ pt_to_func_pred (PARSER_CONTEXT * parser, PT_NODE * spec, PT_NODE * expr)
 static bool
 validate_regu_key_function_index (REGU_VARIABLE * regu_var)
 {
+  /* if add it here, add it to validate_regu_key_function_index () as well */
```
Comment (hgryoo) [inline src/parser/xasl_generation.c:24184]:
```suggestion
  /* if add it here, add it to pt_expr_is_allowed_as_function_index () as well */
```

**Review by hgryoo — COMMENTED** [review]
It seems that the comments are written in reverse.

File: src/parser/parse_tree_cl.c [hunk 2]
```diff
@@ -18355,6 +18355,7 @@ pt_expr_is_allowed_as_function_index (const PT_NODE * expr)
 {
   assert (expr != NULL && expr->node_type == PT_EXPR);
 
+  /* if add it here, add it to pt_expr_is_allowed_as_function_index() as well */
```
Comment (hgryoo) [inline src/parser/parse_tree_cl.c:18358]:
```suggestion
  /* if add it here, add it to validate_regu_key_function_index () as well */
```

**Review by hgryoo — APPROVED** [review]

**Review by byungwook-kim — APPROVED** [review]

**Pull request merged**